api/v1beta1/common_types.go

```diff
@@ -112,120 +112,32 @@ type BarbicanComponentTemplate struct {
-        // A string containing the HSM type (currently supported: "luna").
-        Type string `json:"type"`
-
-        // +kubebuilder:validation:Required
-	// Path to vendor's PKCS11 library
-	LibraryPath string `json:"libraryPath"`
-
-        // +kubebuilder:validation:Optional
-        // Token serial number used to identify the token to be used.
-	// One of TokenSerialNumber, TokenLabels or SlotId must
-	// be defined.  TokenSerialNumber takes priority over
-	// TokenLabels and SlotId
-	TokenSerialNumber string `json:"tokenSerialNumber,omitempty"`
-
-        // +kubebuilder:validation:Optional
-	// Token labels used to identify the token to be used.
-	// One of TokenSerialNumber, TokenLabels or SlotId must
-	// be specified. TokenLabels takes priority over SlotId.
-	// This can be a comma separated string of labels
-	TokenLabels string `json:"tokenLabels,omitempty"`
-
-        // +kubebuilder:validation:Optional
-	// One of TokenSerialNumber, TokenLabels or SlotId must
-	// be defined.  SlotId is used if none of the others is defined
-        SlotId string `json:"slotId,omitempty"`
-
-        // +kubebuilder:validation:Required
-	// Label to identify master KEK in the HSM (must not be the same as HMAC label)
-	MKEKLabel string `json:"MKEKLabel"`
-
-        // +kubebuilder:validation:Optional
-	// +kubebuilder:default=32
-	// Length in bytes of master KEK
-	MKEKLength int `json:"MKEKLength"`
+const (
+	// SecretStoreSimpleCrypto -
+	SecretStoreSimpleCrypto SecretStore = "simple_crypto"
 
-        // +kubebuilder:validation:Required
-        // Label to identify HMAC key in the HSM (must not be the same as MKEK label)
-        HMACLabel string `json:"HMACLabel"`
-
-        // +kubebuilder:validation:Optional
-	// +kubebuilder:default=CKK_GENERIC_SECRET
-	// HMAC Key Type
-	HMACKeyType string `json:"HMACKeyType"`
+	// SecretStorePKCS11 -
+	SecretStorePKCS11 SecretStore = "pkcs11"
 
-        // +kubebuilder:validation:Optional
-	// +kubebuilder:default=CKM_GENERIC_SECRET_KEY_GEN
-	// HMAC Keygen Mechanism
-	HMACKeygenMechanism string `json:"HMACKeygenMechanism"`
-
-        // +kubebuilder:validation:Optional
-	// +kubebuilder:default=CKM_SHA256_HMAC
-	// HMAC Mechanism. This replaces hsm_keywrap_mechanism
-	HMACMechanism string `json:"HMACMechanism"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=4
-        // +kubebuilder:validation:Maximum=7
-        // +kubebuilder:validation:Minimum=0
-        // Level of logging, where 0 means "no logging" and 7 means "debug".
-        LoggingLevel int `json:"loggingLevel"`
-
-	// +kubebuilder:validation:Required
-	// The HSM's IPv4 address (X.Y.Z.K)
-	ServerAddress string `json:"serverAddress"`
-
-	// +kubebuilder:validation:Optional
-	// The IP address of the client connecting to the HSM (X.Y.Z.K)
-	ClientAddress string `json:"clientAddress,omitempty"`
+	// DefaultP11ClientDataPath
+	DefaultP11ClientDataPath = "/etc/hsm-client"
```

Comment:
Nit: Consider renaming this to `DefaultPKCS11ClientDataPath`.  I think we should be consistent with naming variables, and this patch uses both `PKCS11` as in `SecretStorePKCS11` in line 120 above, and also `P11` as used here.  It may not be obvious to other folks that `P11` is short for `PKCS11` and that both will be used in the same backend.  I think it would be better and clearer to just use `PKCS11` throughout.

Reply:
Done